radix-engine/src/state_manager/staging.rs

```diff
@@ -0,0 +1,216 @@
+use sbor::rust::collections::*;
+use sbor::rust::vec::Vec;
+use scrypto::buffer::{scrypto_decode, scrypto_encode};
+
+use crate::ledger::*;
+
+/// Nodes form an acyclic graph towards the parent
+struct StagedExecutionStoreNode {
+    parent_id: u64,
```

Comment:
Out of curiosity, why do we need `parent_id`? Looks like the id is maintained by the owner.

Reply:
Which owner?